packages/compass-connections/src/hooks/use-connection-status.ts

```diff
@@ -0,0 +1,35 @@
+import { useConnectionsManagerContext } from '../provider';
+import {
+  type ConnectionStatus,
+  ConnectionsManagerEvents,
+} from '../connections-manager';
+
+import { type ConnectionInfo } from '@mongodb-js/connection-info';
+import { useEffect, useState } from 'react';
+
+export function useConnectionStatus(connectionInfoId: ConnectionInfo['id']): {
+  status: ConnectionStatus;
+} {
+  const connectionManager = useConnectionsManagerContext();
+  const [status, setStatus] = useState<ConnectionStatus>(
+    connectionManager.statusOf(connectionInfoId)
+  );
+
+  const updateStatus = () => {
+    setStatus(connectionManager.statusOf(connectionInfoId));
+  };
```

Comment:
This needs to be wrapped in useCallback, or you should move the definition inside effect. You are recreating this function every render unnecessarily adding and removing subscriptions from the connection manager instead of doing it once